js/data/data_source/utils.js

```diff
@@ -0,0 +1,133 @@
+import ajax from '../../core/utils/ajax';
+import Store from '../abstract_store';
+import ArrayStore from '../array_store';
+import { each, map } from '../../core/utils/iterator';
+import CustomStore from '../custom_store';
+import { extend } from '../../core/utils/extend';
+import { isPlainObject } from '../../core/utils/type';
+import dataUtils from '../utils';
+
+export const CANCELED_TOKEN = 'canceled';
+
+export const isPending = deferred => deferred.state() === 'pending';
+
+export const normalizeStoreLoadOptionAccessorArguments = (originalArguments) => {
+    switch(originalArguments.length) {
+        case 0:
+            return undefined;
+        case 1:
+            return originalArguments[0];
+    }
+    return [].slice.call(originalArguments);
+};
+
+export const generateStoreLoadOptionAccessor = (optionName) => function() {
```

Comment:
`function()` -> `() => ` and other same places

Reply:
We should return `function` to change context (we use `this`).
Should we pass `this` as `context` to a function?

https://github.com/DevExpress/DevExtreme/pull/11479/files#diff-6b83de546e57031c44bc875e92bbfc24L141-L150

https://github.com/DevExpress/DevExtreme/pull/11479/files#diff-6b83de546e57031c44bc875e92bbfc24L579

Reply:
`export const generateStoreLoadOptionAccessor = (optionName, context) => () => ...`

Reply:
It's bad practice using context inside simple functions.
The best solution is to get rid of using context, but I don't know how to achieve it now.

Reply:
I can try define this function in Data Source class (then we will same context for `this`). Or we can get rid of `generateStoreLoadOptionAccessor` (generator) and use result function in class.